base/shell/explorer/startup.cpp

```diff
@@ -532,15 +539,28 @@ BOOL DoStartStartupItems(ITrayWindow *Tray)
             return FALSE;
     }
 
-    dwWait = WaitForSingleObject(s_hStartupMutex, INFINITE);
+    DWORD dwWait = WaitForSingleObject(s_hStartupMutex, INFINITE);
     TRACE("dwWait: 0x%08lX\n", dwWait);
     if (dwWait != WAIT_OBJECT_0)
     {
         TRACE("LastError: %ld\n", GetLastError());
 
-        DoFinishStartupItems();
+        ReleaseStartupMutex();
         return FALSE;
     }
+    return TRUE;
+}
+
+BOOL DoFinishStartupItems(VOID)
```

Comment:
It doesn't seem useful to make this function return anything.

Reply:
Don't ask me why it has a return value...